### ISSUE TITLE
Fix all CPU routines

### ISSUE THREAD

**Issue by anushkrish** [barbagroup/cuIBM#20]

**Comment by mesnardo** [comment]
Non-functional CPU routines will be removed from the `master` (in-progress).
The code with those routines is now available in the branch called `deprecated-cpu-routines`.
